lizrd/train/train_utils.py

```diff
@@ -52,5 +70,30 @@ def get_model(
     ).to(last_gpu)
 
     model = llm.LLM(embedding_layer, encoder_tower, head)
+    if ddp_enabled:
+        model = wrap_in_ddp(module=model, rank=rank)
+    elif fsdp_enabled:
+        model = wrap_in_fsdp(
+            module=model,
+            rank=rank,
+            param_precision=fsdp_param_precision,
+            cast_inputs=True,
+            mixed_precision_ignored_classes=fsdp_mixed_precision_ignore_classes,
+            offload_params=fsdp_offload_params,
+            print_model=True,
+            min_num_params=fsdp_min_num_params,
+            modules_to_wrap=fsdp_modules_to_wrap,
+            is_logging_process=is_logging_process,
+        )
+
+    if gradient_checkpointing_modules is not None:
+        check_fn = lambda x: isinstance(x, gradient_checkpointing_modules)
+        non_reentrant_wrapper = partial(
+            checkpoint_wrapper,
+            checkpoint_impl=CheckpointImpl.NO_REENTRANT,
+        )
+        apply_activation_checkpointing(
```

Comment:
does it also work without FSDP? The function `apply_activation_checkpointing` comes from torch distributed. If it's not intended to be used without fsdp, please add an assert

Reply:
[this](https://app.neptune.ai/o/pmtest/org/llm-random/runs/table?viewId=99f374ec-39cd-48c4-ac55-0f8f7c35500c&detailsTab=monitoring&shortId=LLMRANDOM-3657&type=run&compare=MwNgrA7ANAjEA&query=(sys%2Fid%3Astring%20%3D%20%22LLMRANDOM-3657%22)%20OR%20(sys%2Fid%3Astring%20%3D%20%22LLMRANDOM-3658%22)&sortBy=%5B%22sys%2Fcreation_time%22%5D&sortFieldType=%5B%22datetime%22%5D&sortFieldAggregationMode=%5B%22auto%22%5D&sortDirection=%5B%22descending%22%5D&suggestionsEnabled=false&lbViewUnpacked=true) proves that it works regardless of fsdp.